### ISSUE TITLE
After Upgrade to 2.2.0 Link redirect to Dashboard

### ISSUE THREAD

**Issue by EchoRPK** [nextcloud/forms#836]
**Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!**

**Describe the bug**

**To Reproduce**
1. Click on 'Forms' Button

**Expected behavior**
Loading the Forms App

**Nextcloud (please complete the following information):**
- Nextcloud-Version: 20.0.8
- Forms-Version: 2.2.0

**Desktop (please complete the following information):**
 - OS: Win 10
 - Browser Chrome, Firefox
 - Version 88, 86

**Comment by EchoRPK** [comment]
Yep Cached Removed. nextcloud log says nothing about forms. Browser says also nothing about Forms. Only a few Bootstrap Errors and:

activity-sidebar.js?v=406e3cfc-15:603 Uncaught TypeError: Cannot read property 'extend' of undefined
    at activity-sidebar.js?v=406e3cfc-15:603
    at activity-sidebar.js?v=406e3cfc-15:803

**Comment by EchoRPK** [comment]
Or do you like the Calender Errors to? 

{"reqId":"nWFtMnylRHNzgLRA3YOz","level":3,"time":"2021-03-10T10:30:28+00:00","remoteAddr":"xxx","user":"xxx","app":"PHP","method":"GET","url":"/ocs/v2.php/search/providers/calendar/search?term=log&from=%2Fsettings%2Fuser","message":{"Exception":"Error","Message":"Undefined index: {http://owncloud.org/ns}owner-principal at /var/www/nextcloud/apps/dav/lib/CalDAV/CalDavBackend.php#1789","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/dav/lib/CalDAV/CalDavBackend.php","line":1789,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/apps/dav/lib/Search/EventsSearchProvider.php","line":116,"function":"searchPrincipalUri","class":"OCA\\DAV\\CalDAV\\CalDavBackend","type":"->"},{"file":"/var/www/nextcloud/lib/private/Search/SearchComposer.php","line":162,"function":"search","class":"OCA\\DAV\\Search\\EventsSearchProvider","type":"->"},{"file":"/var/www/nextcloud/core/Controller/UnifiedSearchController.php","line":110,"function":"search","class":"OC\\Search\\SearchComposer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":169,"function":"search","class":"OC\\Core\\Controller\\UnifiedSearchController","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":100,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":152,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":309,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/ocs/v1.php","line":88,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/ocs/v2.php","line":24,"args":["/var/www/nextcloud/ocs/v1.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":91,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36","version":"20.0.8.1"}

**Comment by jotoeri** [comment]
😄 no, calendar is not of interest here.
Well - for me it works flawless. I just saw the issue for a short moment after update, but the second click worked (which is annoying but ok imo, server/cache might have to realise the new routes). Maybe some php-cache, which was still populated?
I'm sorry, but i have not even one ansatz where to look here. As far as i see, this setup should work, i don't even have an error-message by hand. 🤔 🤷‍♂️

**Comment by EchoRPK** [comment]
Ok after rebooting the server completly ist worked for me ;)

**Comment by tlinna** [comment]
I'd like to revive this. I'm having this same weird problem. Browser gets a 302 response from /apps/forms/ and redirects straight into /apps/dashboard/

Previously shared form link works as expected, so I suppose the app is working fine in the background.

Tried rebooting the server, no help. I have no idea what to try next.

**Comment by jotoeri** [comment]
Did you also clear your Browser-Cache? Do you have some Error-Logs?

Generally - for both of you - Please first give it a try on the forum for such stuff. https://help.nextcloud.com Especially if you don't have a clear Error-Message or clearly identified bug. Providing support is normally not, what should be done on the issue-tracker here.

**Comment by tlinna** [comment]
Yes, cleared cache and tried different browsers (Firefox, Chrome, Edge). No errors in nextcloud log. No errors in Nginx log. This definitely started from the 2.2.0 forms update and only affects the forms app as far as I can see.

Nothing happens on the browser side before the redirect. No idea why it would happen suddenly after 2.2.0 update (we've been using a form from the previous version for a week or so).

Similar 302 redirect to dashboard happens with a nonsense apps url like /apps/asdf/

Here's Nginx (1.18.0) access log for the page load. 

`[10/Mar/2021:13:31:33 +0000] "GET /apps/forms HTTP/2.0" 301 178 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36"
[10/Mar/2021:13:31:33 +0000] "GET /apps/forms/ HTTP/2.0" 302 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36"
[10/Mar/2021:13:31:33 +0000] "GET /apps/dashboard/ HTTP/2.0" 200 11551 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36"
`

Nextcloud 20.0.8 running on Ubuntu Server 20.04 LTS, PHP 7.4.3 and Nginx 1.18.0, using recommended Nginx config from some months ago.

**Comment by birefringence** [comment]
Same here with Nginx 1.14.2, PHP 7.3.27 and Nextcloud 21.0.0.

**Comment by tlinna** [comment]
Just to give an update... After a couple of reboots, couple reinstalls of the app, almost a full day, and preparing for emergency retrieval of responses, it suddenly started working by itself. Still no idea what was wrong or what fixed it. No idea what sort of cache could persist through all that... but sure, it _seems_ like a cache problem 😁 

Sorry for the clutter

**Comment by jotoeri** [comment]
Just for reference - I now created a forum thread for support on this.
https://help.nextcloud.com/t/forms-update-redirect-to-dashboard/110567
